clang/lib/AST/ByteCode/InterpBuiltin.cpp

```diff
@@ -256,15 +256,28 @@ static bool interp__builtin_strlen(InterpState &S, CodePtr OpPC,
     return false;
 
   assert(StrPtr.getFieldDesc()->isPrimitiveArray());
-
+  unsigned ElemSize = StrPtr.getFieldDesc()->getElemSize();
   size_t Len = 0;
   for (size_t I = StrPtr.getIndex();; ++I, ++Len) {
     const Pointer &ElemPtr = StrPtr.atIndex(I);
 
     if (!CheckRange(S, OpPC, ElemPtr, AK_Read))
       return false;
 
-    uint8_t Val = ElemPtr.deref<uint8_t>();
+    uint32_t Val;
```

Comment:
Should we assert that if `ID == Builtin::BIwcslen` ElemSize is the size of wchar_t?